### PR TITLE
Add status command

### PR DESCRIPTION
**Pull request by tomer-stripe**
### Reviewers
r? @ob-stripe @aarongreen-stripe @brandur-stripe @auchenberg-stripe 
cc @stripe/dev-platform

 ### Summary
Let people retrieve API status through the CLI. This pulls from our status site so that is the source of truth.

The command is `stripe status` and includes several flags with:
1. `--verbose`, lists out status of individual systems
2. `--format`, let's people render the status as json
3. `--poll`, continuously checks the status site
4. `--poll-rate`, let's people check more frequently than ever 60 seconds (up to 5 seconds per poll)
5. `--hide-spinner`,  hides the spinner that's shown when polling

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,6 +5,7 @@ go 1.12
 require (
 	github.com/BurntSushi/toml v0.3.1
 	github.com/briandowns/spinner v0.0.0-20190319032542-ac46072a5a91
+	github.com/coreos/etcd v3.3.10+incompatible
```
Comment (ob-stripe) [inline go.mod]: Was this intended? This isn't listed as an indirect dependency, but I can't find it actually being used anywhere.

Comment (tomer-stripe) [reply go.mod]: Hmm, probably not. I'm not sure where that came from to be honest, removing.

**Comment by tomer-stripe** [comment]
@ob-stripe yea that's a good idea. I wonder if we should do something more universal around emoji support that's built as part of the colors flag.

File: pkg/status/status.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+			return "", err
+		}
+
+		return string(data), nil
+	}
+
+	statusString := fmt.Sprintf(`%s %s{{if .statuses}}
+%s API
+%s Dashboard
+%s StripeJS
```
Comment (brandur-stripe) [inline pkg/status/status.go]: Note a big deal, but I wonder if we should use the official product name/case/punctuation here, which I think is "Stripe.js" for StripeJS and just "Checkout" for CheckoutJS.

File: pkg/status/status.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+	Time        string   `json:"time"`
+}
+
+type statuses struct {
+	API        string `json:"api"`
+	Dashboard  string `json:"dashboard"`
+	Stripejs   string `json:"stripejs"`
+	Checkoutjs string `json:"checkoutjs"`
+	Webhooks   string `json:"webhooks"`
+	Emails     string `json:"emails"`
```
Comment (brandur-stripe) [inline pkg/status/status.go]: Not sure if intentional or not, but "emails" doesn't seem to appear in the formatted output below.

File: pkg/status/status.go (hunk 2 above)
Comment (tomer-stripe) [reply pkg/status/status.go]: Ah, good call! I'll change this.

File: pkg/status/status.go (hunk 3 above)
Comment (tomer-stripe) [reply pkg/status/status.go]: I intentionally left it out. We don't report emails on status.stripe.com so I didn't want to cause confusion. Maybe I should remove it from here too?

Comment (brandur-stripe) [reply pkg/status/status.go]: Probably not a big deal either way. If you leave it in, I'd add a comment indicating that it's not currently exposed (and may be not reliable? not sure if there's a reason it's not).

**Pull request merged**